init.sh

```diff
@@ -31,6 +33,33 @@ if [ -z "${module_name}" ]; then
     module_name=$(basename $output_path)
 fi
 
+# check module name using go mod init
+tmp_dir=
+
+cleanup() {
+    cd ${CURR_PWD}
+    if [ ! -z "$tmp_dir" ]; then
+        rm -fr "$tmp_dir"
+    fi
+}
+
+trap cleanup EXIT
+
+tmp_dir=`mktemp -d -q`
+exit_code=$?
+if [ $exit_code -ne 0 ]; then
+    echo "fail to create temporary directory"
+    exit 1
+fi
+
+pushd ${tmp_dir}
+    go mod init ${module_name}
+    if [ $? -ne 0 ]; then
+        exit 1
+    fi
+popd
+rm -fr "$tmp_dir"
```

Comment:
this line looks redundant as cleanup is always triggered

Reply:
done